aspnetcore/fundamentals/middleware/write/6sample/WebMiddleware/Program.cs

```diff
@@ -0,0 +1,72 @@
+#define THIRD // FIRST SECOND THIRD
+#if NEVER
+#elif FIRST
+// <snippet_first>
+using System.Globalization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.UseHttpsRedirection();
+
+app.Use(async (context, next) =>
+{
+    var cultureQuery = context.Request.Query["culture"];
+    if (!string.IsNullOrWhiteSpace(cultureQuery))
+    {
+        var culture = new CultureInfo(cultureQuery);
+
+        CultureInfo.CurrentCulture = culture;
+        CultureInfo.CurrentUICulture = culture;
+    }
+
+    // Call the next delegate/middleware in the pipeline.
+    await next();
```

Comment:
We could encourage the user to pass the context in all code snippets.

At this point it is using the ` Func<Task>` and immediately in the doc it is mentioned that it could be moved to class,

```
public class RequestCultureMiddleware
{
    private readonly RequestDelegate _next;

    public RequestCultureMiddleware(RequestDelegate next)
    {
        _next = next;
    }

    public async Task InvokeAsync(HttpContext context)
    {
        var cultureQuery = context.Request.Query["culture"];
        if (!string.IsNullOrWhiteSpace(cultureQuery))
        {
            var culture = new CultureInfo(cultureQuery);

            CultureInfo.CurrentCulture = culture;
            CultureInfo.CurrentUICulture = culture;
        }

        // Call the next delegate/middleware in the pipeline.
        await _next(context);
    }
}
```

Here, it is using `RequestDelegate` instead of `Func<Task>`. People might get confused.

An one-liner info on the difference between `_next(context)` and `_next()` could be helpful